src/gui/qgstextformatwidget.cpp

```diff
@@ -200,6 +200,10 @@ void QgsTextFormatWidget::initWidget()
   connect( chkLineBelow, &QAbstractButton::toggled, this, &QgsTextFormatWidget::updateLinePlacementOptions );
   connect( chkLineOn, &QAbstractButton::toggled, this, &QgsTextFormatWidget::updateLinePlacementOptions );
 
+  mTextOrientationComboBox->addItem( tr( "Horizontal" ), QgsTextFormat::HorizontalOrientation );
```

Comment:
Could be nice to have icons for these

Reply:
Yes agreed, that's in the to-do list but I'd like to approach this through a series or PRs. It'll hopefully come attached to the next PR.